### PR TITLE
Ability to disable notifications of useCheckAuth and useLogoutIfAccessDenied hooks

### PR DESCRIPTION
**Pull request by WiXSL**
Implements #5178

### CODE REVIEW

**Review by djhi — APPROVED** [review]
:+1:

**Review by Luwangel — COMMENTED** [review]
Great job, thanks! I think it's a good opportunity to write tests for these two hooks. Do you have time to write them?

**Comment by WiXSL** [comment]
@Luwangel I don't have time right now, sorry.

**Comment by WiXSL** [comment]
Done!

File: packages/ra-core/src/auth/useCheckAuth.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                />
+            </AuthContext.Provider>
+        );
+        await wait();
+        expect(logout).toHaveBeenCalledTimes(0);
+        expect(notify).toHaveBeenCalledTimes(0);
+        expect(queryByText('not authenticated')).not.toBeNull();
+    });
+
+    it('should logout whitout showing a notification', async () => {
```
Comment (djhi) [inline packages/ra-core/src/auth/useCheckAuth.spec.tsx:108]:
```suggestion
    it('should logout whitout showing a notification when disableNotification is true', async () => {
```

**Review by djhi — CHANGES_REQUESTED** [review]
Nice!

**Review by Luwangel — APPROVED** [review]
Thanks :)

**Review by djhi — APPROVED** [review]

**Pull request merged**